tests/test.d

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env rdmd
+/**
+Platform independent port of `test.sh`. Runs the tests in this directory.
+
+Ignores differences in line endings, unless the test uses `--end_of_line`.
+**/
+import std.algorithm, std.array, std.conv, std.file, std.path, std.process;
+import std.stdio, std.string, std.typecons, std.range, std.uni;
+
+version (Windows)
+    enum dfmt = `..\bin\dfmt.exe`;
+else
+    enum dfmt = `../bin/dfmt`;
+
+int main()
+{
+    foreach (braceStyle; ["allman", "otbs", "knr"])
+        foreach (entry; dirEntries(".", "*.d", SpanMode.shallow).
+                            filter!(e => e.baseName(".d") != thisExePath.baseName(".exe")))
```

Comment:
I think it would be better to hardcode `test.d` here as it doesn't necessarily need to be run with the same executable name, e.g. it may not match with a temporary executable.

Also can you move the . from above to this line (`.filter`)?